dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java

```diff
@@ -170,20 +167,210 @@ Entry parseWDLContent(Entry entry, String content) {
      */
     public abstract Map<String, String> getWorkflowGitUrl2RepositoryId();
 
-    /**
-     * Given a repositoryid, get a workflow
-     * TODO: pass in an existing workflow when we need to override paths
-     * @param repositoryId uniquely identify a repo
-     * @param existingWorkflow an existing workflow entry, when existingWorkflow is a stub or empty, simply create a new stub entry
-     *                         when existingWorkflow is a full workflow, do a full refresh but use the existing workflow paths as a guide
-     *                         for where to look for files
-     * @return a fully realized workflow
-     */
-    public abstract Workflow getNewWorkflow(String repositoryId, Optional<Workflow> existingWorkflow);
-
     List<String> getWdlImports(File workflowFile){
         Bridge bridge = new Bridge();
         return bridge.getImportFiles(workflowFile);
     }
 
+    /**
+     * Given the content of a file, determines if it is a valid WDL workflow
+     * @param content
+     * @return true if valid WDL workflow, false otherwise
+         */
+    public Boolean checkValidWDLWorkflow(String content) {
```

Comment:
Hmmm, this might harken back to the first comment. If big-B boolean has a special meaning when set to null, document it. Otherwise use a regular boolean.